onnxruntime/core/optimizer/transpose_optimization/onnx_transpose_optimization.cc

```diff
@@ -1363,7 +1544,7 @@ static int EstimateTransposeValueCost(const api::GraphRef& graph, std::string_vi
       if (dq_input_node != nullptr) {
         if (dq_input_node->OpType() == "Squeeze") {
           auto squeeze_input_node = graph.GetNodeProducingOutput(dq_input_node->Inputs()[0]);
-          if (squeeze_input_node->OpType() == "Transpose") {
+          if (squeeze_input_node != nullptr && squeeze_input_node->OpType() == "Transpose") {
```

Comment:
Note: fix nullptr dereference for Squeeze that consumes a graph input (`GetNodeProducingOutput()` returns null). Caught by new unit test TransposeOptimizerTests.CancelSqueezeBeforePerAxisDQ.